lib/jwe/validator.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module JWE
+  # Validates JWE parameters (algorithm, encryption, compression, and key)
+  class Validator
```

Comment:
As the validator has no state I would prefer module and static methods over a class that needs instantiation in this case.